tests/API/Helpers/State/SessionStateHandlerTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace Auth0\Tests\Api\Helpers\State;
+
+use Auth0\SDK\API\Helpers\State\SessionStateHandler;
+use Auth0\SDK\Store\SessionStore;
+
+/**
+ * Class SessionStateHandlerTest
+ *
+ * @package Auth0\Tests\Api\Helpers\State
+ */
+class SessionStateHandlerTest extends \PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Storage engine to use.
+     *
+     * @var SessionStore
+     */
+    private $store;
+
+    /**
+     * State handler to use.
+     *
+     * @var SessionStateHandler
+     */
+    private $state;
+
+    /**
+     * SessionStateHandlerTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        // Suppress header sent error
+        @$this->store = new SessionStore();
+        $this->state = new SessionStateHandler($this->store);
+    }
+
+    /**
+     * Test that state is stored and retrieved properly.
+     */
+    public function testStateStoredCorrectly()
+    {
+        $uniqid = uniqid();
+        $this->state->store($uniqid);
+        $this->assertEquals($uniqid, $this->store->get(SessionStateHandler::STATE_NAME));
```

Comment:
I missed this one before I guess.. why if you haven't passed a state name above, you're passing one now? Either pass always the name or never.

Reply:
I think you're misreading, these are 2 different methods. I'll name for clarity.